plugins/gstperf.c

```diff
@@ -198,13 +235,73 @@ gst_perf_get_property (GObject * object, guint property_id,
   }
 }
 
+static gboolean
+gst_perf_update_bps (void *data)
+{
+  guint buffer_current_idx;
+  GstPerf *perf;
+
+  g_return_val_if_fail(data, FALSE);
+
+  perf = GST_PERF (data);
+
+  /* Calculate bits per second */
+  perf->bps =
+      perf->byte_count * GST_PERF_BITS_PER_BYTE / (perf->bps_frequency / 1000.0);
```

Comment:
You need to serialize *every* access to `byte_count` and `byte_count_total`. Actually, everything that is modified in the `transform` should be protected.

Reply:
`bps_frequency` can be changed at runtime by the user. The callback frequency and the calculation will no longer match.

Reply:
Also, protect access to `bps_frequency` as well. User's can modify the property at any time and this occurs in a different thread.